manifests/tasks/sanitise-object-name.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sanitise-object-name
+spec:
+  description: This Task can be used to sanitise a string that will be used for naming a k8s object. This task converts the string to a value that can be used for an object name. Any non-alphanumeric characters that are converted to "-" 
+  params:
+  - name: input-string
+    type: string
+  results:
+  - name: output-string
+    description: Sanitised output string
+  steps:
+  - name: sanitise-input-string
+    image: registry.access.redhat.com/ubi9/ubi-minimal
+    script: |
+      #!/usr/bin/env bash
+
+      set -Eeuo pipefail
+
+      echo $(params.input-string) | sed 's/[^a-z0-9.-]/-/g; s/-*$//' | tr -d "\n"  > $(results.output-string.path);
```

Comment:
There are some other possible checks we might want to do in future to the proposed name before returning, now that this is a generic task dedicated to that purpose.

E.g. check length and shorten if too long (or maybe error if empty?)

Reply:
Also the end currently just checks to make sure it's not a `-`, but I guess it also needs to check that it's not a `.`, right? Similarly the start of the string should be checked at that point for both characters, since they are invalid starting characters, and even though the input string may not have begun with them, we may have dropped other invalid characters preceding them in the first sed expression, right?

Also, maybe the expected behaviour for A-Z chars is that they're translated to a-z, rather than dropped.
E.g.:
- MY-MODEL-123 -> my-model-123 (rather than just 123)
- My-MoDeL-123 -> my-model-123 (rather than y-oe-123)

Thoughts?

Reply:
Yeah I think that all makes sense the point of checking if `-` is at the start is a good catch because that isn't valid either.

And I think change the A-Z to lowercase is a good idea I will try and figure that part out now